cli/EAP7-943_Security_Auth_Commands.adoc

```diff
@@ -0,0 +1,241 @@
+
+== Overview
+
+Extend the existing CLI security command with a set of actions to configure 
+elytron authentication.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3672
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-943
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+CLI
+
+=== Other Interested Projects
+
+elytron
+
+== Requirements
+
+* Offer simple actions to enable/disable HTTP and SASL authentication for both 
+Http server and management interfaces.
+* Hide complexity of elytron management model as much as possible without introducing 
+unwanted simplification.
+* Take benefit of existing Out Of The Box elytron configuration.
+* Drive users to select the right set of options thanks to CLI completion that 
+controls visibility of options.
+* Help users discover the underlying elytron model thanks to the echo-dmr command 
+to print out the set of sent model operations.
+* Allow user to visualise the management or application authentication configuration.
+
+=== Hard Requirements
+
+* For SASL supported mechanisms are EXTERNAL, DIGEST-MD5. PLAIN, JBOSS-LOCAL-USER, SCRAM-*, DIGEST-SHA-*
+* For HTTP supported mechanisms are CLIENT-CERT, BASIC, DIGEST and FORM
+
+==== Configure SASL Authentication for management interfaces
+ 
+==== enable-sasl-management
+Associate an elytron SASL factory to the management interface. In case a SASL factory already exists,
+ the factory is extended (or updated) with the selected mechanism. 
+If no mechanism is provided, the out of the box SASL factory is associated to the management interface. 
+This command creates all the non existing resources required to configure authentication. 
+This command can be called multiple times to configure the referenced SASL factory.
+ 
+   
+_security enable-sasl-management ( [--file-system-realm=<a string>]
+                                    [--user-role-decoder=<a string>] |
+                                    [--user-properties-file=<null>]
+                                    [--group-properties-file=<null>]
+                                    [--relative-to=<a string>] )
+                                    [--mechanism=<a string>]
+                                    [--exposed-realm-name=<a string>]
+                                    [--external-trust-store=<a string>]
+                                    [--management-interface=<a string>]
+                                    [--new-auth-factory-name=<a string>]
+                                    [--new-realm-name=<a string>]
+                                    [--new-security-domain-name=<a string>]
+                                    [--no-reload] [--super-user]_
+ 
+ 
+The mininal command to associate the out of the box elytron SASL factory to the HTTP interface +
+_security enable-sasl-management_
+ 
+The mininal command to enable EXTERNAL:+
```

Comment:
A space should probably be before '+' char, like on line 84 ( https://github.com/jfdenise/wildfly-proposals/blob/master/cli/EAP7-943_Security_Auth_Commands.adoc )